hpke.go

```diff
@@ -256,12 +311,29 @@ func (ctx *EncryptContext) Seal(aad, pt []byte) []byte {
 	return ct
 }
 
+func (ctx *EncryptContext) Marshal() ([]byte, error) {
+	return syntax.Marshal(ctx.inner)
+}
+
+func UnmarshalEncryptContext(opaque []byte) (*EncryptContext, error) {
+	ctx, err := unmarshalCipherContext(opaque)
+	if err != nil {
+		return nil, err
+	}
+
+	if ctx.inner.Role != roleSender {
```

Comment:
If we're validating the role, should we also validate that the export secret and nonce are the right length according to the AEAD and KDF IDs? (I think the key length is taken care of when we try to initialize an AEAD at L256.)

Reply:
Good idea. The exporter secret length should equal the output size of the Extract, correct?

Reply:
Yep! It's `Nh`, which is the output size of `Extract`.